component/azstorage/datalake.go

```diff
@@ -58,10 +58,11 @@ import (
 
 type Datalake struct {
 	AzStorageConnection
-	Auth       azAuth
-	Service    *service.Client
-	Filesystem *filesystem.Client
-	BlockBlob  BlockBlob
+	Auth           azAuth
+	Service        *service.Client
+	Filesystem     *filesystem.Client
+	BlockBlob      BlockBlob
+	datalakeCPKOpt *directory.CPKInfo
```

Comment:
Please add changelog and tests for the datalake APIs where CPK is added.

Reply:
I have added them pls review